packet/routes/admin.py

```diff
@@ -0,0 +1,43 @@
+from flask import render_template
+
+from packet import app
+from packet.models import Packet, Freshman
+from packet.routes.shared import packet_sort_key
+from packet.utils import before_request, packet_auth, admin_auth
+from packet.log_utils import log_cache, log_time
+
+
+@app.route('/admin/packets')
+@log_cache
+@packet_auth
+@admin_auth
+@before_request
+@log_time
```

Comment:
This is getting to be a lot of decorators. I'd consider refactoring the auth and logging ones to be function-style decorators so you can make things more compact and readable. Maybe something like this:
```python
@app.route('/admin/packets')
@with_logs(TIME, CACHE)
@require_auth(ADMIN, PACKET)
@before_request
```